apps/meteor/app/lib/server/functions/sendMessage.js

```diff
@@ -269,13 +270,19 @@ export const sendMessage = async function (user, message, room, upsert = false)
 			// so, we don't really care if it is successful or fails
 			void Apps.getBridges()?.getListenerBridge().messageEvent('IPostMessageSent', message);
 		}
-
 		/*
 		Defer other updates as their return is not interesting to the user
 		*/
 
 		// Execute all callbacks
 		await callbacks.run('afterSaveMessage', message, room);
+		if (
+			user._id === room.v._id &&
+			isOmnichannelRoom(room) &&
+			(room.verificationStatus === 'isListeningToEmail' || room.verificationStatus === 'isListeningToOTP')
+		) {
+			await callbacks.run('verificationCheck', room, message.msg);
```

Comment:
Instead of creating this new callback, I think we can re-use the existing "afterSaveMessage" callback and check from these room conditions over there